File: 0682/Sesión 04 - lib java/Publicaciones_v02/src/main/java/pe/edu/pucp/inf30/sesiones/publicaciones02/publicaciones/estandardAPA/Tesis.java
package pe.edu.pucp.inf30.sesiones.publicaciones02.publicaciones.estandardAPA;

import pe.edu.pucp.inf30.sesiones.publicaciones02.publicaciones.Clase_de_Tesis;
import pe.edu.pucp.inf30.sesiones.publicaciones02.publicaciones.Publicacion;

public class Tesis extends Publicacion {
    
    private Clase_de_Tesis clase;
    private String departamento;
    private String institucion;
    private String ciudad;
    private String estado;

    public Tesis(String nombre, String nombre_completo_autor, Integer año, Clase_de_Tesis clase, String departamento, String institucion, String ciudad, String estado) {
        super(nombre, nombre_completo_autor, año);
        this.clase = clase;
        this.departamento = departamento;
        this.institucion = institucion;
        this.ciudad = ciudad;
        this.estado = estado;
    }

    public Tesis(String nombre, String nombres_completos_autores[], Integer año, Clase_de_Tesis clase, String departamento, String institucion, String ciudad, String estado) {
        super(nombre, nombres_completos_autores, año);
        this.clase = clase;
        this.departamento = departamento;
        this.institucion = institucion;
        this.ciudad = ciudad;
        this.estado = estado;
    }

    @Override
    public String toString() {
        String referencia = "";
        for (int i = 0; i < cantidad_autores; i++) {
            if (i != 0) {
                if (i == cantidad_autores - 1) {
                    referencia = referencia.concat(" and ");
                } else {
                    referencia = referencia.concat(", ");
                }
            }
            referencia = referencia.concat(autores.get(i).toString());
        }
        referencia = referencia.concat(". (");
        referencia = referencia.concat(año.toString());
        referencia = referencia.concat("). ");        
        referencia = referencia.concat(nombre);
        referencia = referencia.concat(" [");
        referencia = switch (clase) {
            case DOCTORADO -> referencia.concat("Tesis de Doctorado");
            case MAESTRIA -> referencia.concat("Tesis de Maestría");
            default -> referencia.concat("Tesis de Licenciatura");
        };
        referencia = referencia.concat(", ");
        referencia = referencia.concat(departamento);
        referencia = referencia.concat(", ");
        referencia = referencia.concat(institucion);
        referencia = referencia.concat(", ");
        referencia = referencia.concat(ciudad);
        referencia = referencia.concat(", ");
        referencia = referencia.concat(estado);        
        referencia = referencia.concat("].");
        return referencia;
    }
}

File: 0682/Sesión 05 - lib C#/Publicaciones_v02/Publicaciones_v02/Publicaciones/EstandardAPA/Tesis.cs
using Publicaciones;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Publicaciones.EstandardAPA
{
    internal class Tesis: Publicacion
    {        
        private Clase_de_Tesis clase;
        private string departamento;
        private string institucion;
        private string ciudad;
        private string estado;
   
        public Tesis(string nombre, string nombre_completo_autor, int año, Clase_de_Tesis clase, string departamento, string institucion, string ciudad, string estado) 
            : base(nombre, nombre_completo_autor, año)
        {
            this.clase = clase;
            this.departamento = departamento;
            this.institucion= institucion;
            this.ciudad = ciudad;
            this.estado = estado;
        }

        public Tesis(string nombre, string[] nombres_completos_autores, int año, Clase_de_Tesis clase, string departamento, string institucion, string ciudad, string estado) 
            : base(nombre, nombres_completos_autores, año)
        {
            this.clase = clase;
            this.departamento = departamento;
            this.institucion = institucion;
            this.ciudad = ciudad;
            this.estado = estado;
        }

        public override string ToString()
        {
            string referencia = "";
            for (int i = 0; i < this.autores.Count; i++)
            {
                if (i != 0)
                {
                    if (i == this.cantidad_autores - 1)
                        referencia += " and ";
                    else
                        referencia += ", ";
                }
                referencia += this.autores[i].ToString();
            }
            referencia += ". (";
            referencia += this.año;
            referencia += "). ";
            referencia += this.nombre;
            referencia += ". [";
            switch (this.clase)
            {
                case Clase_de_Tesis.DOCTORADO: referencia += "Tesis de Doctorado"; break;
                case Clase_de_Tesis.MAESTRIA: referencia += "Tesis de Maestría"; break;
                case Clase_de_Tesis.LICENCIATURA: referencia += "Tesis de Licenciatura"; break;
            }
            referencia += ", ";
            referencia += this.departamento;
            referencia += ", ";
            referencia += this.institucion;
            referencia += ", ";
            referencia += this.ciudad;
            referencia += ", ";
            referencia += this.estado;
            referencia += "].";
            return referencia;
        }
    }
}
